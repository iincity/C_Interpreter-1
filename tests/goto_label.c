#include <stdio.h>


int main()
{
	int i, j = 3;

start1:
	i = j;

	int a = 5;
start2:
	printf("%d\n", i);
	--i;
	a--;
	printf("%d\n", a);
	if (i)
		goto start2;

	printf("%d\n", j);
	if (--j)
		goto start1;


	printf("%d\n", i);
	printf("%d\n", j);
	printf("%d\n", a);


	int b = 0;
	goto into;

	{
		int b = 4;

into:
		++b;
		printf("%d\n", b);
		int a = b / 2;
		printf("%d\n", a);
	}

	{
		int b = 3;
		printf("%d\n", b);

		goto outof;

		int a = b++;
		printf("%d\n", a);
		printf("%d\n", b);
	}

outof:
	printf("%d\n", b);


//goto endblock;  <--- illegal/non-standard pointless

	{
		int i = 100;
		printf("%d\n", i);
//endblock:
	}


goto beginblock;

	j += 10;
	printf("%d\n", j);

beginblock:
	{
		int j = 200;
		printf("%d\n", j);
	}


	{
		int a = 10;
		printf("%d\n", a);
		{
			goto sib_inner;
			{
				printf("%d\n", a);
			}
			int a = 5;
			printf("%d\n", a);
		}

		printf("%d\n", a);
	
		{
			int a = 15;
			{
sib_inner:
				printf("%d\n", a);

			}
		}
	}

	goto decl;

	{
		int a = 10;
		{
			int a = 5;
			printf("%d\n", a);
decl:
			;    // <-- label can only go with a statement not a declaration
			int b = 6;
			printf("%d\n", b);

		}
	}

	int check = 2;	
	{
		int test = 321;
		{
backinto:
			printf("%d\n", test);

		}
		printf("%d\n", test);
		test /= 2;
		check--;
	}

	int test = 123;
	printf("%d\n", test);
	
	if (check)
		goto backinto;


	return 0;
}
	
