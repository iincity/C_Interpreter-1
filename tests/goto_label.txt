
int main()
{
	int i, j = 3;

start1:
	i = j;

	int a = 5;
start2:
	print i;
	i -= 1;
	a -= 1;
	print a;
	if (i)
		goto start2;

	print j;
	j -= 1;
	if (j)
		goto start1;


	print i;
	print j;
	print a;

	int b = 0;
	goto into;

	{
		int b = 4;

into:
		b += 1;
		print b;
		int a = b / 2;
		print a;
	}

	{
		int b = 3;
		print b;

		goto outof;

		int a = b += 1;
		print a;
	}

outof:
	print b;



//goto endblock;  <--- illegal/non-standard pointless

	{
		int i = 100;
		print i;
//endblock:
	}


goto beginblock;

	j += 10;
	print j;

beginblock:
	{
		int j = 200;
		print j;
	}


	{
		int a = 10;
		print a;
		{
			goto sib_inner;
			{
				print a;
			}
			int a = 5;
			print a;
		}

		print a;
	
		{
			int a = 15;
			{
sib_inner:
				print a;

			}
		}
	}

	goto decl;

	{
		int a = 10;
		{
			int a = 5;
			print a;
decl:
			;    // <-- label can only go with a statement not a declaration
			int b = 6;
			print b;

		}
	}

	int check = 2;	
	{
		int test = 321;
		{
backinto:
			print test;

		}
		print test;
		test /= 2;
		check -= 1;
	}

	int test = 123;
	print test;
	
	if (check)
		goto backinto;



	return 0;
}
	
