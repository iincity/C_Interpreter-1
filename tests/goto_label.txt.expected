3
4
2
3
1
2
3
2
4
1
3
2
1
4
1
0
0
4
1
0
3
0
100
200
10
0
6
321
321
123
0
0
123
